File: src/types/license.ts
export interface License {
  id: string;
  license_key: string;
  customer_name?: string;
  license_type: string;
  status: string;
  issued_at: string;
  activated_at: string;
  expires_at?: string;
  created_at: string;
  updated_at: string;
}

export interface LicenseInfo {
  has_license: boolean;
  license_type?: string;
  customer_name?: string;
  status?: string;
  expires_at?: string;
  days_remaining?: number;
  is_trial: boolean;
}

export interface GeneratedKey {
  key: string;
  license_type: string;
  expires_at: string | null;
}

File: src-tauri/src/models/license.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct License {
    pub id: String,
    pub license_key: String,
    pub customer_name: Option<String>,
    pub license_type: String,
    pub status: String,
    pub issued_at: String,
    pub activated_at: String,
    pub expires_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct LicenseInfo {
    pub has_license: bool,
    pub license_type: Option<String>,
    pub customer_name: Option<String>,
    pub status: Option<String>,
    pub expires_at: Option<String>,
    pub days_remaining: Option<i64>,
    pub is_trial: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ValidationLog {
    pub id: String,
    pub license_id: Option<String>,
    pub success: bool,
    pub error_message: Option<String>,
    pub validated_at: String,
    pub app_version: Option<String>,
}

/// Risultato della generazione chiave
#[derive(Debug, Serialize)]
pub struct GeneratedKey {
    pub key: String,
    pub license_type: String,
    pub expires_at: Option<String>,
}
